boards/PDM/Inc/Io/Io_Efuse.h

```diff
@@ -0,0 +1,145 @@
+ */
+#pragma once
+
+#include <stdint.h>
+#include <stdbool.h>
+#include "main.h"
+#include "App_SharedExitCode.h"
+
+struct Efuse_Context;
+
+/**
+ * Allocate and initialize an Efuse
+ * @param hspi Handle to the SPI peripheral used for the Efuse
+ * @param chip_select_port Handle to Efuse's chip-select GPIO port
+ * @param chip_select_port Handle to Efuse's chip-select GPIO pin
+ * @param fsob_port Handle to Efuse's fail-safe output GPIO port
+ * @param fsob_pin Handle to Efuse's fail-safe output GPIO pin
+ * @param fsb_port Handle to Efuse's fault status GPIO port
+ * @param fsb_pin Handle to Efuse's fault status GPIO pin
+ * @param channel0_port Handle to Efuse's channel 0 GPIO port
+ * @param channel0_pin Handle to Efuse's channel 0 GPIO pin
+ * @param channel1_port Handle to Efuse's channel 1 GPIO port
+ * @param channel1_pin Handle to Efuse's channel 1 GPIO pin
+ * @return The created Efuse, whose ownership is given to the caller
+ */
+struct Efuse_Context *Io_Efuse_Create(
+    SPI_HandleTypeDef *hspi,
+    GPIO_TypeDef *     chip_select_port,
+    uint16_t           chip_select_pin,
+    GPIO_TypeDef *     fsob_port,
+    uint16_t           fsob_pin,
+    GPIO_TypeDef *     fsb_port,
+    uint16_t           fsb_pin,
+    GPIO_TypeDef *     channel0_port,
+    uint16_t           channel0_pin,
+    GPIO_TypeDef *     channel1_port,
+    uint16_t           channel1_pin);
+
+/**
+ * Deallocate the memory used by the given Efuse
+ * @param e_fuse The Efuse to deallocate
+ */
+void Io_Efuse_Destroy(struct Efuse_Context *e_fuse);
+
+/**
+ * Enable the channel 0 output
+ */
+void Io_Efuse_Channel0Enable(const struct Efuse_Context *const e_fuse);
+
+/**
+ * Disable the channel 0 output
+ */
+void Io_Efuse_Channel0Disable(const struct Efuse_Context *const e_fuse);
+
+/**
+ * Enable the channel 1 output
+ */
+void Io_Efuse_Channel1Enable(const struct Efuse_Context *const e_fuse);
+
+/**
+ * Disable the channel 1 output
+ */
+void Io_Efuse_Channel1Disable(const struct Efuse_Context *const e_fuse);
+
+/**
+ * Set the current/temperature monitoring option of CSNS pin
+ * @param selection The desired monitoring function which can be selected by
+ * specifying the following values:
+ *                  CSNS_FUNCTION_DISABLED - Current sensing disabled
+ *                  CSNS_FUNCTION_CURRENT_CH0 - Current sensing for Channel 0
+ *                  CSNS_FUNCTION_CURRENT_CH1 - Current sensing for Channel 1
+ *                  CSNS_FUNCTION_TEMPERATURE - Efuse temperature sensing
+ *                  CSNS_FUNCTION_CURRENT_SUM - Current sensing for summed
+ * channels (for Parallel mode)
+ * @param e_fuse Pointer to the Efuse structure for the Efuse being configured
+ * @return EXIT_CODE_OK if the channel configuration was successful
+ *         EXIT_CODE_TIMEOUT if one of the SPI writes timed-out
+ */
+ExitCode Io_Efuse_ConfigureChannelMonitoring(
+    uint8_t                     selection,
+    struct Efuse_Context *const e_fuse);
```

Comment:
can we change `e_fuse` to `efuse` here just to stay consistent with the rest of the files

Reply:
changed